crates/wasmtime/src/store.rs

```diff
@@ -1320,7 +1383,12 @@ impl StoreOpaque {
         // to clean up this fiber. Do so by raising a trap which will
         // abort all wasm and get caught on the other side to clean
         // things up.
-        unsafe { self.async_cx().block_on(Pin::new_unchecked(&mut future)) }
+        unsafe {
+            match self.async_cx() {
+                None => Err(Trap::new("attempted to pull async context during shutdown")),
```

Comment:
Like the above for `Func::new` I think this is safe to `.expect()` and/or `.unwrap()` since it should be impossible to hit this with a `None`

Reply:
Addressed in a775e97.